common/log/loggerimpl/logger.go

```diff
@@ -82,20 +106,25 @@ func caller(skip int) string {
 }
 
 func (lg *loggerImpl) buildFieldsWithCallat(tags []tag.Tag) []zap.Field {
-	fs := lg.buildFields(tags)
+	fs := lg.buildFields(tags, "")
 	fs = append(fs, zap.String(tag.LoggingCallAtKey, caller(lg.skip)))
 	return fs
 }
 
-func (lg *loggerImpl) buildFields(tags []tag.Tag) []zap.Field {
+func (lg *loggerImpl) buildFields(tags []tag.Tag, keyPrefix string) []zap.Field {
 	fs := make([]zap.Field, 0, len(tags))
 	for _, t := range tags {
 		f := t.Field()
 		if f.Key == "" {
 			// ignore empty field(which can be constructed manually)
 			continue
 		}
+		f.Key = keyPrefix + f.Key
 		fs = append(fs, f)
+
+		if withTags, ok := f.Interface.(taggable); ok {
+			fs = append(fs, lg.buildFields(withTags.Tags(), fmt.Sprintf("%s-", f.Key))...)
```

Comment:
```suggestion
			fs = append(fs, lg.buildFields(withTags.Tags(), f.Key + "-"))...)
```
no need to pass it through an extremely-general formatter when it's already a string